### PR TITLE
add guide on post build hook

### LINKED ISSUE #463: Nix post-build hooks
Migrate `post-build-hook` guide to nix.dev. This is a valuable guide that doesn't belong to the Nix reference manual, but should absolutely be preserved and maintained.

More information: https://github.com/NixOS/nix/pull/7979

Note: This is currently [blocked on license incompatibility of LGPL and CC-BY-SA](https://github.com/NixOS/nix/pull/7979#issuecomment-1460860939).

Blocked by: https://github.com/NixOS/nix.dev/issues/478

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
Closes #463

originally written under LGPL 2.1 for the Nix reference manual.
authors as recorded in the Nix repository:

    git log --pretty="Co-authored-by: %an <%ae>" -- doc/manual/advanced-topics/post-build-hook.xml doc/manual/src/advanced-topics/post-build-hook.md | sort | uniq

relicenced with authors' permission to CC-BY-SA 4.0: https://github.com/NixOS/nix.dev/issues/463

### CODE REVIEW

**Comment by infinisil** [comment]
Now that the CLA is gone, it seems fine to proceed with this. @fricklerhandwerk sound good?

**Comment by fricklerhandwerk** [comment]
Yeah, it just needs to be done. Free for grabs really, my queue is way too long to ever terminate anyway.

**Comment by proofconstruction** [comment]
Rebased this and fixed some of the links (and the errors due to the vale check I didn't notice before), but now linkcheck fails in an unrelated file due to https://github.com/NixOS/nixos-homepage/issues/1448; pausing this while that's sorted.

File: source/guides/recipes/post-build-hook.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+```
+
+Then update [`nix.conf`](https://nixos.org/manual/nix/command-ref/conf-file.html) on any machine that will access the cache.
+Add the cache URL to [`substituters`](https://nixos.org/manual/nix/command-ref/conf-file.html#conf-substituters) and the public key to [`trusted-public-keys`](https://nixos.org/manual/nix/command-ref/conf-file.html#conf-trusted-public-keys):
+
+    substituters = https://cache.nixos.org/ s3://example-nix-cache
+    trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= example-nix-cache-1:1/cKDz3QCCOmwcztD2eV6Coggp6rqc9DGjWv7C0G+rM=
+
+Machines that build for the cache must sign derivations using the private key.
+On those machines, add the path to the key file to the [`secret-key-files`](https://nixos.org/manual/nix/command-ref/conf-file.html#conf-secret-key-files) field in their [`nix.conf`](https://nixos.org/manual/nix/command-ref/conf-file.html):
```
Comment (fricklerhandwerk) [inline source/guides/recipes/post-build-hook.md:38]:
```suggestion
On those machines, add the path to the file with the private key to the [`secret-key-files`](https://nixos.org/manual/nix/command-ref/conf-file.html#conf-secret-key-files) field in their [`nix.conf`](https://nixos.org/manual/nix/command-ref/conf-file.html):
```

File: source/guides/recipes/post-build-hook.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    secret-key-files = /etc/nix/key.private
+
+We will restart the Nix daemon in a later step.
```
Comment (fricklerhandwerk) [inline source/guides/recipes/post-build-hook.md:42]:
```suggestion
You will restart the Nix daemon in a later step.
```

**Pull request merged**